### PR TITLE
[fix] remove `_test_type_and_finiteness` method from `LinearRegression` and `LogisticRegression`

### PR DESCRIPTION
**Pull request by icfaust**
### Description 
This PR removes the _type_and_finiteness method from LinearRegression and LogisticRegression. This checks for finiteness and if it is a complex valued object.  The finite check was duplicated for the predict method in these estimators, and in cases that sklearnex fell back to sklearn for evaluation. In both of these estimators (in sklearn) require finiteness to operate, and is not necessary for the onedal_support check. The estimator should fail just as is the case for sklearn.

Complex numbers are not supported in sklearn or sklearnex, so it should just fail at a standard check_array or validate_data call.  This call can also be removed (not sure why it was included in the first place).

Therefore the entire check can be removed from both estimators.

This was first found and discussed in #1843.

Fixes # - _issue number(s) if exists_

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.
- [ ] I have resolved any merge conflicts that might occur with the base branch.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [ ] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Review by ethanglaser — APPROVED** [review]
LGTM, assuming green internal CI

**Comment by samir-nasibli** [comment]
Thank you!
Transparent description and justification in the description provided 👍
Looks go to me as well. Assuming green CI

**Review by samir-nasibli — APPROVED** [review]

**Comment by icfaust** [comment]
Private CI run for verification before merge: http://intel-ci.intel.com/ef4262c1-99e0-f13e-98a1-a4bf010d0e2e

**Review by DDJHB — APPROVED** [review]
ltgm, thanks for detailed description
